pkg/csilvm/server.go

```diff
@@ -126,18 +110,179 @@ func (s *Server) GetPluginInfo(
 	return response, nil
 }
 
-// ControllerService RPCs
+func (s *Server) GetPluginCapabilities(
+	ctx context.Context,
+	request *csi.GetPluginCapabilitiesRequest) (*csi.GetPluginCapabilitiesResponse, error) {
+	if err := s.validateGetPluginCapabilitiesRequest(request); err != nil {
+		return nil, err
+	}
+	response := &csi.GetPluginCapabilitiesResponse{
+		Capabilities: []*csi.PluginCapability{
+			{
+				Type: &csi.PluginCapability_Service_{
+					Service: &csi.PluginCapability_Service{
+						Type: csi.PluginCapability_Service_CONTROLLER_SERVICE,
+					},
+				},
+			},
+		},
+	}
+	return response, nil
+}
 
-func (s *Server) ControllerProbe(
+// Probe initializes the Server by creating or opening the VolumeGroup. The
+// sanity checks and initialization logic are idempotent to allow this call to
+// be performed after initialization.
+func (s *Server) Probe(
```

Comment:
This function is exactly the same as the old `NodeProbe` which was simply renamed.

Reply:
Pretty sure that this probe could be executed more frequently since in later versions of CSI this is considered to be more of a plugin "health check"

Reply:
Hmm you're right. For now I've updated the comment but are you suggesting I move the initialization logic into the plugin's startup sequence?

Reply:
I think this logic might make sense as part of the startup sequence vs. Probe. If the startup sequence fails, then the program can either crash (read: fail-fast, preferred) or can linger and just return unhealthy (less preferred).